src/algebraic_geometry/prime_spectrum.lean

```diff
@@ -0,0 +1,188 @@
+    all_goals
+    { rw mem_zero_locus at h ⊢,
+      have hx : ideal.span _ ≤ x.as_ideal := ideal.span_le.mpr h,
+      refine set.subset.trans _ hx,
+      intros r hr, cases hr, assumption } },
+  { rintro h,
+    rw set.mem_union,
+    simp only [mem_zero_locus] at h ⊢,
+    rw classical.or_iff_not_imp_right,
+    intros hI r hr,
+    rw set.not_subset at hI,
+    rcases hI with ⟨s, hs1, hs2⟩,
+    apply (ideal.is_prime.mem_or_mem (by apply_instance) _).resolve_left hs2,
+    apply h,
+    rw [submodule.mem_coe, submodule.mem_inf],
+    split,
+    { exact ideal.mul_mem_left _ (ideal.subset_span hr) },
+    { exact ideal.mul_mem_right _ (ideal.subset_span hs1) } }
+end
+
+lemma zero_locus_Union {ι : Type*} (I : ι → set R) :
+  zero_locus (⋃ i, I i) = (⋂ i, zero_locus (I i)) :=
+by { ext x, simp only [mem_zero_locus, set.mem_Inter, set.Union_subset_iff] }
+
+lemma Inter_zero_locus {ι : Type*} (I : ι → set R) :
+  (⋂ i, zero_locus (I i)) = zero_locus (⋃ i, I i) :=
+(zero_locus_Union I).symm
+
+/-- The Zariski topology on the prime spectrum of a commutative ring
+is defined via the closed sets of the topology:
+they are exactly those sets that are the zero locus of a subset of the ring. -/
+instance Zariski_topology : topological_space (prime_spectrum R) :=
+topological_space.of_closed {Z | ∃ I, Z = prime_spectrum.zero_locus I}
+(⟨set.univ, by simp⟩)
+begin
+  intros Zs h,
+  rw set.sInter_eq_Inter,
+  let f : Zs → set R := λ i, classical.some (h i.2),
+  have hf : ∀ i : Zs, i.1 = zero_locus (f i) := λ i, classical.some_spec (h i.2),
+  simp only [hf],
+  exact ⟨_, Inter_zero_locus _⟩
+end
+(by { rintro _ _ ⟨I, rfl⟩ ⟨J, rfl⟩, exact ⟨_, union_zero_locus I J⟩ })
+
+lemma is_open_iff (U : set (prime_spectrum R)) :
+  is_open U ↔ ∃ I, -U = zero_locus I :=
+iff.rfl
+
+lemma is_closed_iff_zero_locus (Z : set (prime_spectrum R)) :
+  is_closed Z ↔ ∃ I, Z = zero_locus I :=
+by rw [is_closed, is_open_iff, set.compl_compl]
+
+lemma zero_locus_is_closed (I : set R) :
+  is_closed (zero_locus I) :=
+by { rw [is_closed_iff_zero_locus], exact ⟨I, rfl⟩ }
+
+section comap
+variables {S : Type v} [comm_ring S] {S' : Type*} [comm_ring S']
+
+/-- The function between prime spectra of commutative rings induced by a ring homomorphism.
+This function is continuous. -/
+def comap (f : R →+* S) : prime_spectrum S → prime_spectrum R :=
+λ y, ⟨ideal.comap f y.as_ideal, by { refine ideal.is_prime.comap _, apply_instance }⟩
+
+variables (f : R →+* S)
+
+@[simp] lemma comap_as_ideal (y : prime_spectrum S) :
+  (comap f y).as_ideal = ideal.comap f y.as_ideal :=
+rfl
+
+@[simp] lemma comap_id : comap (ring_hom.id R) = id :=
+funext $ λ x, ext.mpr $ by { rw [comap_as_ideal], apply ideal.ext, intros r, simp }
+
+@[simp] lemma comap_comp (f : R →+* S) (g : S →+* S') :
+  comap (g.comp f) = comap f ∘ comap g :=
+funext $ λ x, ext.mpr $ by { simp, refl }
+
+@[simp] lemma preimage_comap_zero_locus (I : set R) :
+  comap f ⁻¹' (zero_locus I) = zero_locus (f '' I) :=
```

Comment:
```suggestion
  (comap f) ⁻¹' (zero_locus I) = zero_locus (f '' I) :=
```
Did I parse correctly the statement? The parenthesis are not necessary, but they would help here.

Reply:
Thanks, I agree that this improves readability.